### DISCUSSION TITLE
UnicodeEncodeError:

### DISCUSSION THREAD

**Discussion by Dabnabbit** [sylikc/pyexiftool#51] [Q&A]
![image](https://user-images.githubusercontent.com/8258356/175803265-8c124141-ee7d-43e7-984e-a5e582efcad2.png)
I've been struggling to get this module to run, was reading through the FAQ/troubleshooting and haven't seen anything like this, but it returns this same error on any file targeted.

`UnicodeEncodeError: 'charmap' codec can't encode character '\x81' in position 18: character maps to <undefined>`

I'm running exiftool ver 12.42, and pyexiftool 0.5.3 installed via PIP; I've enabled debug logging but can't seem to figure it out, it seems like a byte code translation error of some kind, but would love some suggestions.

**Comment by sylikc** [comment]
Hi, I've been a bit backed up in real life, but I think I just need to roll in this change https://github.com/sylikc/pyexiftool/pull/48 .  I wanted to do more 'testing' per se, but I think it'll fix the problem you're having here.  Can you test that branch and see if it'll fix your problem?

It's likely that some tag in that jpg file is not UTF-8... ``raw_bytes`` option will allow it to return bytes and not crash.

**Reply by Dabnabbit** [reply]
Hey thanks for the quick reply! That is very likely the case, as I know from experience that many exif metadata tags, especially in some Samsung phone cameras, store the "Date Taken" field (and a few others) with additional Unicode characters: [char]8206, [char]8207, [char]8234, and [char]8236 (invisible text alignment characters); This has caused issues with other string parsers I've used in the past and/or written in PowerShell to interact directly with the windows "Details" attributes tagged on each file. 

That said after reviewing your changes in #48, I'd say that is very likely the culprit when you removed the -encoding tag from the popen(). I don't think I'll be able to test out of the branch this week to confirm that fixes it, however, it's extremely likely the culprit, and I'd be surprised to see that there are not more images being affected by it.

**Reply by sylikc** [reply]
I won't be able to merge this this week, but next week I should get a chance to.  I think it'll fix the problem... if you can review any potential impact it may have on your code, please do.  Otherwise, I'll merge it as is when I have time.  Thanks
